### ISSUE TITLE
Fields and stamping

### ISSUE THREAD

**Issue by baylesa-dev** [pdfcpu/pdfcpu#929]
Hi there!

Thanks so much for the amazing work you're doing here!

I've been trying for a week to insert fields (checkboxes) into a PDF through the API. I tried creating a new PDF without a background and stamping it over my source PDF, but it seems the fields aren't "stamped."

Is there a way to "stamp" form fields with pdfcpu?

Thanks a lot!

**Comment by hhrutter** [comment]
If you generate a PDF form then you'd want to fill the form fields rather than stamp them.

Check out the `pdfcpu form fill`  command (`pdfcpu help form`) on the CLI and corresponding tests in `form_test.go`.

**Comment by GitHubRulesOK** [comment]
Hmmm it is common for users to want to "ADD" form fields and widgets via command line onto an existing PDF 
Agreed this is exceptionally hard as inserting /Annot [1  0 R 2 0 R 3 0 R] etc. into an **existing page header** and thus done by very very few SDKs

There was one in the past that allowed for command line insertions
```
Add a new form field  
Add a new form at page 1, coordinates (50/50), width = 200, height = 25,  font = 4, 
font size = 10 and write "Hello World!" into the form. 
pdform in.pdf out.pdf +field02@1,50,50,250,75,4,10="Hello World!" 
```

The OP has seen that if they are referenced in a new blank PDF page that may be overstamped onto the existing page (Again few libraries would attempt that).

**Comment by GitHubRulesOK** [comment]
There should in theory be no problem "overstamping"  or "blending" the background of a form and the foreground fields as long as there are not nominal overlapping objects.

Here is a classic textual form (with no binary contents) and the complex fields as drawn in another page overlay if you blend page the single pages or overprint and dont allow clashing it should work same as the target file format.

[FormSheet.pdf](https://github.com/user-attachments/files/17124918/FormSheet.pdf)
[FormFields.pdf](https://github.com/user-attachments/files/17124919/FormFields.pdf)

Currently as a stamp command nothing is imported if we try to place fields on sheet  
`pdfcpu stamp add -mode pdf -- "formfields.pdf" "" formsheet.pdf out.pdf`

HOWEVER if we use the logic that a form can underlay a field and use sheet to stamp the fields,. Then they are mixed with the sheet. I forgot that pdfcpu will try to use stamps as watermarks (a very odd assumption).  
`pdfcpu stamp add -mode pdf -- "formsheet.pdf" "" formfields.pdf out.pdf`

So it half works and perhaps with a bit of fettling work well  
![image](https://github.com/user-attachments/assets/e4d5e9b0-6705-4b9a-b083-dbbc146dd3d0)

**Comment by GitHubRulesOK** [comment]
@baylesa-dev
PDf logic is often stack-wise annd that works in this case create a blank page with your forms data fields as they are unrelated to any other page (just their own relative page positions). Then using stamp without the default rotation/scale position the form visuals as required. (Form fields are never directly attributable to any page graphics). Form Fields as "annotation widgets will always float "ontop".

**Comment by hhrutter** [comment]
Hello!

pdfcpu maintainer here replying directly to @baylesa-dev 

It depends on what it really is you are trying to do:

A) You have a PDF that is missing smth that looks like a checkbox
and you do not need a PDF form, meaning no real interactivity required?

B) Your input file is really a PDF form that is missing a checkbox?

C) Your input file needs a form containing a checkbox that may be checked later on by somebody else?

Once I fully understand your intentions I may help you.

**Comment by hhrutter** [comment]
@GitHubRulesOK 

>>  I forgot that pdfcpu will try to use stamps as watermarks (a very odd assumption).

Please carefully read the docs at pdfcpu.io before you start ranting:
https://pdfcpu.io/core/stamp
https://pdfcpu.io/core/watermark

It's all about drawing content before the original page content or after.
Drawing content before the original page content creates the impression of a watermark.
Drawing content after the original page content creates the impression of a stamp.

**Comment by GitHubRulesOK** [comment]
@hhrutter 
it was not intended as a "rant" I spend a lot of time promoting pdfcpu as having powerful upcoming command line solutions to older issues. 
My observation was that a stamp from say qpdf would be expected to be upright and scaled around fitting existing centre page

**Comment by hhrutter** [comment]
Fair enough -  but If you think there may be a related issue please do me a favor and create a new one instead of posting obeservations below an existing issue.

Sometimes the pdfcpu discussion boards may also be the better place to put these.

This will not only help me triaging but also user who would like to see simple on point answers to an issue.

Thanks for spreading the word 💚
